src/include/hardware.h

```diff
@@ -169,15 +169,15 @@
 /*
  * Atomic versions of the flag set/clear macros.
  */
-#define FLD_ISSET_ATOMIC_16(field, mask) ((field) & (uint16_t)(mask))
+#define FLD_ISSET_ATOMIC_16(field, mask) (__wt_atomic_load16(&(field)) & (uint16_t)(mask))
```

Comment:
GCC has an atomic builtin __atomic_and_fetch, does that work here?

Reply:
Nope, because `and_fetch` applies `and` to `field` before returning the result. I ran into the same issue when first writing this.